src/reaper_osara.cpp

```diff
@@ -1298,7 +1302,19 @@ LRESULT CALLBACK keyboardHookProc(int code, WPARAM wParam, LPARAM lParam) {
 	} else if (wParam == VK_F6 && !(lParam & 0x80000000)) {
 		if (maybeSwitchToFxPluginWindow())
 			return 1;
-	}
+	}else if(wParam == VK_HOME || wParam == VK_END || wParam == VK_SHIFT){
+		//Pressing home / end keys in midi events view moves the cursor to top / bottom respectively, by passing the needed key events to the listview
+		if(wcscmp(className, L"SysListView32") == 0){
+			HWND parent = GetParent(focus);
+			WCHAR parentClassName[22] = L"\0";
+			GetClassNameW(parent, parentClassName, ARRAYSIZE(parentClassName));
+			if(wcscmp(parentClassName, L"REAPERmidieditorwnd") == 0){
+				int msg = WM_KEYDOWN;
+				if((lParam >> 31) == 1) msg=WM_KEYUP;
+				SendMessage(focus, msg, wParam, 0);
+			}//if reaper midi editor window is foreground
```

Comment:
If we're consuming the key, we should return 1 so it doesn't get passed further. We don't want to do this for shift, though, since that might be shift + something else.

Reply:
Thanks for pointing that out! I will fix it if we will continue on this PR.